build/root/Makefile.generated_files

```diff
@@ -434,14 +438,22 @@ $(foreach dir, $(OPENAPI_DIRS), $(eval             \
 ))
 
 # How to regenerate open-api code.  This emits a single file for all results.
-$(OPENAPI_OUTFILE): $(OPENAPI_GEN)
+# The Make rule fails if generated API rule violation report differs from the checked-in
+# violation file, and prints error message to request developer to fix either the API
+# source code, or the known API rule violation file.
+$(OPENAPI_OUTFILE): $(OPENAPI_GEN) $(KNOWN_VIOLATION_FILENAME)
 	./hack/run-in-gopath.sh $(OPENAPI_GEN)                                          \
 	    --v $(KUBE_VERBOSE)                                                         \
 	    --logtostderr                                                               \
 	    -i $$(echo $(addprefix $(PRJ_SRC_PATH)/, $(OPENAPI_DIRS)) | sed 's/ /,/g')  \
 	    -p $(PRJ_SRC_PATH)/$(OPENAPI_OUTPUT_PKG)                                    \
 	    -O $(OPENAPI_BASENAME)                                                      \
-	    "$$@"
+	    -h $(BOILERPLATE_FILENAME)                                                  \
+	    -r $(REPORT_FILENAME)                                                       \
+	    "$$@";                                                                      \
+	diff $(REPORT_FILENAME) $(KNOWN_VIOLATION_FILENAME) ||                          \
```

Comment:
I feel this should be a hack/verify-* thing. Is this being discussed? we are doing verify work in makefile which is odd.

Reply:
We integrated the API rule checking into OpenAPI spec generation (in kube-openapi) to provide an "API compiler / toolchain". In Kubernetes we add an extra layer to allow checking existing violation exceptions, but I think we still want new violations to fail in the "compiling" step. If we move the diff to `hack/verify-openapi-spec.sh` we will have a standalone `hack/update-openapi-spec.sh` script that passes on new violations (although our CI ensures verification scripts always get run, so violations will get caught eventually).